src/battle_script_commands.c

```diff
@@ -16524,6 +16525,18 @@ void BS_TryDefog(void)
     }
 }
 
+void BS_TryUpperHand(void)
+{
+    NATIVE_ARGS(const u8 *failInstr);
+
+    if (GetBattlerTurnOrderNum(gBattlerAttacker) > GetBattlerTurnOrderNum(gBattlerTarget) 
+	|| IS_MOVE_STATUS(gBattleMons[gBattlerTarget].moves[gBattleStruct->chosenMovePositions[gBattlerTarget]]) 
```

Comment:
You need some check here to see if the target actually chose a move

Reply:
Actually, is this not covered by GetChosenMovePriority? Not choosing a move should I assume return 0.

Reply:
> What would the condition for that look like?

You can use `gChosenMoveByBattler`

> Not choosing a move should I assume return 0.

Don't assume. You don't know how the end user will modify the function or how things will change in future generations. The function may end up returning some other number for MOVE_NONE when it should not be. Eg. Say there's an Ability that's added that gives priority to all Normal-type moves. MOVE_NONE may be given priority even though it's not actually a move.